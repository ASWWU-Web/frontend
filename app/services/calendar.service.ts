import { Http } from '@angular/http';
import { Injectable } from '@angular/core';
import 'rxjs/add/operator/map';

@Injectable()
export class CalendarService {
  
  // constructor(private http: Http) {}
  
  // getCalJSON() {
  //   return this.http.get(`https://content.googleapis.com/calendar/v3/calendars/3p9fpi388oqq94pqvm5a2nusp4%40group.calendar.google.com/events?maxResults=10&orderBy=startTime&singleEvents=true&key=AIzaSyDawSineZAnnuDKMQgHiaVt6KuRe4xAzAw`)
  //     .map(response => response.json());
  // }

  
}