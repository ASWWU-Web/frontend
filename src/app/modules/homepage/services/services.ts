export * from './calendar.service';
export * from './super-homepage-search.service';
