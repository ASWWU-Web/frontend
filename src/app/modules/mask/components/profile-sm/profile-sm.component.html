<a *ngIf="this.year == this.current_year" (click)="ballotModal.open(searchResult.username, this.year)">
  <div class="circle">
    <img class="img-fluid" src="{{link}}" alt="{{searchResult.full_name}}"/>
  </div>
  <h4 class=" text-black">{{searchResult.full_name | unescape}}</h4>
</a>

<a *ngIf="this.year != this.current_year" (click)="ballotModal.open(searchResult.username, (this.year) ? this.year: this.current_year)">
  <div class="circle">
    <img class="img-fluid" src="{{link}}" alt="{{searchResult.full_name}}"/>
  </div>
  <h4 id="full-name">{{searchResult.full_name | unescape}}</h4>
</a>

<app-profile-modal #ballotModal></app-profile-modal>
