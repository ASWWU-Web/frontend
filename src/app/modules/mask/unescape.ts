// from: https://stackoverflow.com/questions/47378033/what-the-right-way-unescape-html-entities-in-angular/47386232#47386232

import { Pipe, PipeTransform } from '@angular/core';

@Pipe({
  name: 'unescape'
})
export class UnescapePipe implements PipeTransform {

  transform(value: any, args?: any): any {
    const doc = new DOMParser().parseFromString(value, 'text/html');
    return doc.documentElement.textContent;
  }
}
