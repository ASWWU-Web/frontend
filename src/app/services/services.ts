export * from './calendar.service';
export * from './super-duper.service';
