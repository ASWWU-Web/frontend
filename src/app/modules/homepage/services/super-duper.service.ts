import { Injectable } from '@angular/core';

import { Observable } from 'rxjs/internal/Observable';
import {of, forkJoin} from 'rxjs';
import {map, catchError} from 'rxjs/operators';

import { RequestService } from '../../../../shared-ng/services/services';
import { CURRENT_YEAR } from '../../../../shared-ng/config';

@Injectable()
export class SuperDuperService {

  jobsUri = '/forms/job/view/all';
  maskUri = '/mask/search/' + CURRENT_YEAR + '/';
  pagesUri = '/pages/search?general=';

  jobLink = 'https://aswwu.com/jobs/submit/';
  maskLink = 'https://aswwu.com/mask/profile/';
  pageLink = 'https://aswwu.com/pages/';

  constructor(private requests: RequestService) { }

  private parseQuery(query: string): string {
    // TODO: switch regex with the method used in pages search component
    query = query.toLowerCase().replace(/[!@#$%^&*()=+]/g, '').replace(/[_ -]/g, '%20');
    return query;
  }

  private filterJobs(query, item) {
    query = query.toLowerCase();
    const visible = item.visibility;
    const inDescription = false; // item.job_description.search(query) !== -1;
    const inName = item.job_name.toLowerCase().search(query) !== -1;
    const show = visible && (inDescription || inName);
    return show;
  }

  private mapMask( response ) {
    const toReturn = response.results.slice(0, 3).map(result => {
      return {
        'main': result.full_name,
        'sub': '', // we could add a sub text for each result, like major for mask results
        'source': 'Mask',
        'link': this.maskLink + result.username,
        'top': false,
      };
    });
    toReturn[0].top = true;
    return toReturn;
  }

  private mapJobs( query, response ) {
    const toReturn = response.forms.filter(item => this.filterJobs(query, item)).slice(0, 3).map(result => {
      return {
        'main': result.job_name,
        'sub': '',
        'source': 'Jobs',
        'link': this.jobLink + result.jobID,
        'top': false,
      };
    });
    toReturn[0].top = true;
    return toReturn;
  }

  private mapPages( response ) {
    const toReturn = response.results.slice(0, 3).map(result => {
      return {
        'main': result.title,
        'sub': '',
        'source': 'Pages',
        'link': this.pageLink + result.url,
        'top': false,
      };
    });
    toReturn[0].top = true;
    return toReturn;
  }

  SearchAndReturnObservableArray(query) {
    const queryUri = this.parseQuery(query);
    const maskObservable = this.requests.get( this.maskUri + queryUri ).pipe(
      map(response => this.mapMask(response)),
      catchError(err => of([]))
      );
    const pagesObservable = this.requests.get( this.pagesUri + queryUri ).pipe(
      map(response => this.mapPages(response)),
      catchError(err => of([]))
      );
    const jobsObservable = this.requests.get( this.jobsUri ).pipe(
      map(response => this.mapJobs(query, response)),
      catchError(err => of([]))
      ); // don't parseQuery()
    // uses: https://stackoverflow.com/questions/44141569/how-to-concat-two-observable-arrays-into-a-single-array
    const toReturn = forkJoin(maskObservable, pagesObservable, jobsObservable).pipe(
      map(([s1, s2, s3]) => {
      return [...s1, ...s2, ...s3];
      })
    );
    return toReturn;
  }

  SearchAndReturnObservableResults(query) {
    const queryUri = this.parseQuery(query);
    const maskObservable = this.requests.get( this.maskUri + queryUri ).pipe(
      map(response => response.results),
      catchError(err => of([]))
      );
    const pagesObservable = this.requests.get( this.pagesUri + queryUri ).pipe(
      map(response => response.results),
      catchError(err => of([]))
      );
    const jobsObservable = this.requests.get( this.jobsUri ).pipe(
      map(response => response.forms.filter(result => this.filterJobs(query, result))),
      catchError(err => of([]))
      ); // don't parseQuery()
    // let jobsObservable = this.requests.getObservable( this.jobsUri ).map(response => this.mapJobs(query, response)).catch(err => of([])); // don't parseQuery()

    // uses: https://stackoverflow.com/questions/44141569/how-to-concat-two-observable-arrays-into-a-single-array
    const toReturn = forkJoin(maskObservable, pagesObservable, jobsObservable);
    return toReturn;
  }


}
