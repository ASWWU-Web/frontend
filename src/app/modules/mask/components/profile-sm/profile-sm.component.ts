import { HttpErrorResponse } from '@angular/common/http';
import { Component, Input } from '@angular/core';
import { MEDIA_SM, DEFAULT_PHOTO, CURRENT_YEAR } from '../../../../../shared-ng/config';
import { RequestService } from '../../../../../shared-ng/services/services';

@Component({
  selector: 'profile-sm',
  templateUrl: 'profile-sm.component.html',
  styleUrls: [ 'profile-sm.styles.css' ],
  inputs: ['searchResult']
})


export class ProfileSmComponent {
  link: string = "";

  constructor(private rs: RequestService) {}

  ngOnInit() {
    this.link = this.getPhotoLinkSync(this.searchResult.photo);
    this.getPhotoLinkAsync(this.searchResult.photo)
      .then((link: string) => this.link = link);
  }

  @Input() searchResult: any;
  @Input('year') year: String = undefined;
  current_year = CURRENT_YEAR;

  // Photo url to link function returns proper url and BLANK photo if photo == "None"
  getPhotoLinkSync(url: string) {    
    var link = "";
    if (url && url != 'None') {
      link = MEDIA_SM + '/' + url;
    } else {
      link = MEDIA_SM + '/' + DEFAULT_PHOTO;
    }
    return link;
  }
  // Photo url to link function returns proper url and BLANK photo if photo == "None"
  getPhotoLinkAsync(url: string) {
    return new Promise((resolve, reject) => {      
      var link = this.getPhotoLinkSync(url);
      
      this.rs.get(link).subscribe(
          (data: any) => {
          resolve(link);
        },
        (response: HttpErrorResponse) => { // why does requesting an image resource always respond with error?
          if (response.status == 415) {
            link = MEDIA_SM + '/' + DEFAULT_PHOTO;
            resolve(link);
          } else if (response.status == 200) {
            resolve(link);
          } else {
            reject(response);
          }
        }
      )
    })
  }
}
