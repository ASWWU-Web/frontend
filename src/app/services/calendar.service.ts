import { Injectable } from '@angular/core';

import { RequestService } from 'src/shared-ng/services/services';
import 'rxjs/add/operator/toPromise';

@Injectable()
export class CalendarService {

  private calendarID = '3p9fpi388oqq94pqvm5a2nusp4@group.calendar.google.com';
  private maxResults = 10;
  private orderBy = 'startTime';
  private singleEvents = true;
  private key = 'AIzaSyDawSineZAnnuDKMQgHiaVt6KuRe4xAzAw';

  private upcomingsUrl = 'https://content.googleapis.com/calendar/v3/calendars/' + this.calendarID + '/events?' +
   'maxResults=' + this.maxResults + '&orderBy=' + this.orderBy + '&singleEvents=' + this.singleEvents + '&timeMin=' +
    this.timeMin() + '&key=' + this.key;

  constructor(private rs: RequestService) {  }

  public getUpcomings() {
    return this.rs.get(this.upcomingsUrl).toPromise();
  }

  private timeMin() {
    return new Date().toISOString();
  }
}
