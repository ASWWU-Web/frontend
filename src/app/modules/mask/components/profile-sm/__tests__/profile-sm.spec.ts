import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { ProfileSmComponent } from '../profile-sm.component';
import { UnescapePipe } from '../../../unescape';
import { RequestService } from 'src/shared-ng/services/services';
import { Component } from '@angular/core';

class MockRequestService {
}

@Component({ selector: 'app-profile-modal', template: '' })
class ProfileModalStubComponent {	
}

describe('ProfileSmComponent', () => {
  let component: ProfileSmComponent;
  let fixture: ComponentFixture<ProfileSmComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
	  declarations: [ProfileSmComponent, UnescapePipe, ProfileModalStubComponent],
	  providers: [{ provide: RequestService, useClass: MockRequestService }]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ProfileSmComponent);
    component = fixture.componentInstance;
  });

  it('Should display a flight', () => {
	component.searchResult = { full_name: 'tester', username: 'tester' }
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});