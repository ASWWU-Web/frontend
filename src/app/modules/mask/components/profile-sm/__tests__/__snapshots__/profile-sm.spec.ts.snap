// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileSmComponent Should display a flight 1`] = `
<profile-sm
  current_year={[Function String]}
  link={[Function String]}
  rs={[Function MockRequestService]}
  searchResult={[Function Object]}
  year="undefined"
>
  <a>
    <div
      class="circle"
    >
      <img
        alt="tester"
        class="img-fluid"
        src="https://localhost:8888/media/img-sm/images/default_mask/default.jpg"
      />
    </div>
    <h4
      id="full-name"
    >
      tester
    </h4>
  </a><app-profile-modal />
</profile-sm>
`;
